enterprise/admin/admin.py

```diff
@@ -51,12 +52,25 @@ class EnterpriseCustomerAdmin(SimpleHistoryAdmin):
     EXPORT_AS_CSV_FIELDS = ["name", "active", "site", "uuid"]
 
     actions = [
-        export_as_csv_action("CSV Export", fields=EXPORT_AS_CSV_FIELDS)
+        export_as_csv_action("CSV Export", fields=EXPORT_AS_CSV_FIELDS),
+        get_clear_catalog_id_action()
     ]
 
+    form = EnterpriseCustomerAdminForm
+
     class Meta(object):
         model = EnterpriseCustomer
 
+    def get_form(self, request, obj=None, **kwargs):
+        """
+        Retrieve the appropriate form to use for this admin panel.
+
+        Save the request user into the form for use in loading catalog details.
+        """
+        form = super(EnterpriseCustomerAdmin, self).get_form(request, obj, **kwargs)
+        form.user = request.user
```

Comment:
@haikuginger maybe `user` should become form constructor parameter, as it won't be of much use without it anyway.